browser_history/__init__.py

```diff
@@ -77,3 +82,63 @@ def get_bookmarks():
             utils.logger.info("%s", e)
     output_object.bookmarks.sort()
     return output_object
+
+
+# keep everything lower-cased
+browser_aliases = {
+    "google-chrome": browsers.Chrome,
+    "chromehtml": browsers.Chrome,
+    "chromiumhtm": browsers.Chromium,
+    "chromium-browser": browsers.Chromium,
+    "msedgehtm": browsers.Edge,
+    "operastable": browsers.Opera,
+    "opera-stable": browsers.Opera,
+    "operagxstable": browsers.OperaGX,
+    "firefoxurl": browsers.Firefox,
+    "bravehtml": browsers.Brave,
+}
+
+
+def default_browser():
+    """This method gets the default browser of the current platform
+
+    :return: A :py:class:`browsers.Browser` object representing the default
+        browser in the current platform. If platform is not supported or
+        default browser is unknown or unsupported ``None`` is returned
+
+    :rtype: union[:py:class:`browsers.Browser`, None]
+    """
+    plat = utils.get_platform()
+
+    # ---- get default from specific platform ----
+
+    if plat == utils.Platform.LINUX:
+        default = webbrowser.get().name.lower()
+    elif plat == utils.Platform.WINDOWS:
+        reg_path = (
+            "Software\\Microsoft\\Windows\\Shell\\Associations\\"
+            "UrlAssociations\\https\\UserChoice"
+        )
+        with OpenKey(HKEY_CURRENT_USER, reg_path) as key:
+            default = QueryValueEx(key, "ProgId")[0].lower()
+    else:
+        utils.logger.warning("Default browser feature not supported on this OS")
+        return None
+
+    # ---- convert obtained default to something we understand ----
+
+    b_map = {browser.__name__.lower(): browser for browser in get_browsers()}
+    if default in b_map:
+        # we are lucky and the name is exactly like we want it
+        return b_map[default]
+
+    for browser in browser_aliases:
+        # look for alias matches even if the default name has "noise"
+        # for instance firefox on windows returns something like
+        # "firefoxurl-3EEDF34567DDE" but we only need "firefoxurl"
+        if browser in default:
+            return browser_aliases[browser]
```

Comment:
This might cause issues if the browser_aliases are expanded in the future by someone else. For example, adding `opera` as a key would match `operagx` also. Maybe add a warning near `browser-aliases`?

Reply:
Yeah. I was actually thinking of letting the browser implementations provide their aliases through an `aliases` attribute which will be a list of all aliases for the browser to simplify adding new browsers.

Reply:
That sounds good, having aliases in the browser classes would be better for maintainability.

Direct matching is also a good idea, but might fail in some cases. For example, if there are `operagx` and `opera-browser` keys, and the browser name is `opera` it might match `operagx`. But I understand that this might require some advanced techniques (maybe some NLP?), so checking for a direct match is good enough for now.

Reply:
I solved the issue using two separate passes (loops). First loop checks for direct matches first therefore, `operagx` will be matched with `operagx` first no matter what.
The next loop now checks for rough matches. Even though `operagx` here would match `opera` (the check will assume `gx` is just "noise"), it will have already be matched correctly in the first loop so it won't even get to this point.
NLP would be overkill but good; still, I think python's `difflib` could do the same job of checking for similarity.

Reply:
That looks good. It's probably better to leave enhancements such as using `difflib` for a future PR. The current implementation seems quite good! Just needs more testing from the community.

I'll try to finish the tests soon.